### PR TITLE
add resnet tensorflow model

### PR DESCRIPTION
**Pull request by utkobe**

### CODE REVIEW

File: test/pipelines/object-detection.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (yorkie) [inline test/pipelines/object-detection.json:1]: This needs to be removed?

File: test/pipelines/mnist-image-classification.json [hunk 2]
```diff
@@ -29,4 +29,4 @@
       "package": "./packages/plugins/model-evaluate/image-classification-tfjs-model-evaluate"
     }
   }
-}
+}
```
Comment (yorkie) [inline test/pipelines/mnist-image-classification.json:32]: This change is unnecessary.

File: test/pipelines/databinding-image-classification.json [hunk 3]
```diff
@@ -12,20 +12,18 @@
...
     "modelTrain": {
-      "package": "./packages/plugins/model-train/image-classification-tfjs-model-train",
+      "package": "./packages/plugins/model-train/image-classification-tensorflow-model-train",
       "params": {
-        "epochs": 15
+        "epochs": 20
       }
-    },
-    "modelEvaluate": {
-      "package": "./packages/plugins/model-evaluate/image-classification-tfjs-model-evaluate"
```
Comment (yorkie) [inline test/pipelines/databinding-image-classification.json:28]: The model evaluate should not be removed from our testing pipeline.

File: packages/plugins/model-train/image-classification-tensorflow-model-train/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    let trainDataSet = await createDataset(trainLoader, metadata.labelMap);
+    const count = await trainLoader.len();
+
+    trainDataSet = trainDataSet.repeat();
+    trainDataSet = trainDataSet.batch(batchSize);
+    trainDataSet = trainDataSet.prefetch(boa.kwargs({
+      buffer_size: AUTOTUNE
+    }));
+
+    const trainConfig: any = {
```
Comment (yorkie) [inline packages/plugins/model-train/image-classification-tensorflow-model-train/src/index.ts:63]:
```suggestion
    const trainConfig = {
```

File: packages/plugins/model-train/image-classification-tensorflow-model-train/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  let image = tf.io.read_file(path);
+  image = tf.image.decode_jpeg(image, boa.kwargs({
+    channels: 3
+  }));
+  return image;
+}
+
+async function createDataset(dataLoader: ImageDataLoader, labelMap: {
+  [key: string]: number;
+}) {
```
Comment (Txiaozhe) [inline packages/plugins/model-train/image-classification-tensorflow-model-train/src/index.ts:18]: May be make this as type 'LabelMapType: Record<...>' is better?

Comment (yorkie) [reply packages/plugins/model-train/image-classification-tensorflow-model-train/src/index.ts:18]: In detailed, it's `Record<string, number>`.

**Comment by Txiaozhe** [comment]
There are many warnings like ' <sth> is defined but never used' still, could you please fix it?

File: packages/plugins/model-define/tensorflow-resnet-model-define/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    const log = JSON.parse(fs.readFileSync(path.join(recoverPath, 'log.json'), 'utf8'));
+    labelMap = log.metadata.labelMap;
+    outputShape = Object.keys(labelMap).length;
+  }
+
+  let model: any;
+  model = ResNet50(boa.kwargs({
+    include_top: true,
+    weights: pretrained ? 'imagenet' : null,
+    input_shape:  inputShape,
```
Comment (Txiaozhe) [inline packages/plugins/model-define/tensorflow-resnet-model-define/src/index.ts:62]: space redundancy:(

File: packages/plugins/model-train/image-classification-tensorflow-model-train/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+      buffer_size: AUTOTUNE
+    }));
+
+    const trainConfig: any = {
+      epochs: epochs,
+      steps_per_epoch: parseInt(String(count / batchSize))
+    };
+
+    if (validationLoader) {
+      console.log('create validation dataset');
```
Comment (Txiaozhe) [inline packages/plugins/model-train/image-classification-tensorflow-model-train/src/index.ts:69]: 'console.log' is inappropriate:(

File: packages/plugins/model-train/image-classification-tensorflow-model-train/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+      await trainModel.save_weights(modelPath);
+    });
+
+    const result: TfJsLayersModel = {
+      ...model,
+      model: trainModel
+    };
+    return result;
+  } catch (err) {
+    console.log('[ERROR] model trainer', err);
```
Comment (Txiaozhe) [inline packages/plugins/model-train/image-classification-tensorflow-model-train/src/index.ts:91]: ditto:(

**Comment by Txiaozhe** [comment]
LGTM if nits fixed:)

**Comment by yorkie** [comment]
@utkobe Is this ready for review and merge?

**Comment by utkobe** [comment]
@yorkie @Txiaozhe  Not yet, I'm still testing my different datasets. This time I would like to make sure the model is ready to use to get a good result. In the meantime, I will fix the suggestions from you guys. Thanks

**Comment by Txiaozhe** [comment]
> @yorkie @Txiaozhe Not yet, I'm still testing my different datasets. This time I would like to make sure the model is ready to use to get a good result. In the meantime, I will fix the suggestions from you guys. Thanks

I like it:)

File: packages/plugins/model-train/image-classification-tensorflow-model-train/src/index.ts (hunk 7 above)
Comment (utkobe) [reply packages/plugins/model-train/image-classification-tensorflow-model-train/src/index.ts:69]: what the reason for this. What should be used instead

**Comment by utkobe** [comment]
This is ready

File: packages/plugins/model-train/image-classification-tensorflow-model-train/src/index.ts [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+      await trainModel.save_weights(modelPath);
+    });
+
+    const result: UniModel = {
+      ...model,
+      model: trainModel
+    };
+    return result;
+  } catch (err) {
+    console.log('[ERROR] model trainer', err);
```
Comment (yorkie) [inline packages/plugins/model-train/image-classification-tensorflow-model-train/src/index.ts:96]:
```suggestion
    console.error('occurs an error on model trainer', err);
```

**Comment by utkobe** [comment]
> LGTM but a small nit :)

done

**Pull request merged**